File: webmvp/src/lib/userDisplay.test.ts
import { describe, expect, it } from "vitest";

import { userDisplayName, userInitials } from "./userDisplay";

describe("userInitials", () => {
  it("returns ? for empty names", () => {
    expect(userInitials("")).toBe("?");
    expect(userInitials("   ")).toBe("?");
  });

  it("uses first two letters for single word", () => {
    expect(userInitials("Alex")).toBe("AL");
  });

  it("uses first letters of first two words", () => {
    expect(userInitials("Alex Rivera")).toBe("AR");
  });

  it("handles extra whitespace", () => {
    expect(userInitials("  Morgan   Phillips  ")).toBe("MP");
  });
});

describe("userDisplayName", () => {
  it("prefers display name", () => {
    expect(userDisplayName("Alex Rivera", "alex@example.com")).toBe(
      "Alex Rivera",
    );
  });

  it("falls back to email local part", () => {
    expect(userDisplayName(null, "alex@example.com")).toBe("alex");
  });

  it("returns Guest when nothing provided", () => {
    expect(userDisplayName(null, null)).toBe("Guest");
  });

  it("ignores blank display names", () => {
    expect(userDisplayName("   ", "alex@example.com")).toBe("alex");
  });
});

File: mobile/test/domain/user_display_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:lf_chords/domain/user_display.dart';

void main() {
  test('userInitials', () {
    expect(userInitials('Alex Rivera'), 'AR');
    expect(userInitials('Madonna'), 'MA');
  });

  test('userDisplayName', () {
    expect(userDisplayName('Alex', 'a@b.com'), 'Alex');
    expect(userDisplayName(null, 'alex@example.com'), 'alex');
    expect(userDisplayName(null, null), 'Guest');
  });
}
